src/components/App/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react"
+import { Theme } from "@artsy/palette"
+import { UserContext, AnonymousUser, User } from "components/App/UserContext"
+import { isBrowser } from "utils/environment"
+
+export class App extends React.Component<any, { user: User }> {
+  state = { user: AnonymousUser }
+
+  componentDidMount() {
+    if (isBrowser()) {
+      const user: User = window.localStorage.getItem("gatsbyUser")
+        ? JSON.parse(window.localStorage.getItem("gatsbyUser") as string)
+        : AnonymousUser
+      this.setUser(user)
+    }
+  }
+
+  setUser = (user: User) => {
+    window.localStorage.setItem("gatsbyUser", JSON.stringify(user))
+    this.setState({ user })
+  }
+
+  logout = () => {
+    this.setUser(AnonymousUser)
+  }
+
+  render() {
+    console.log(this.props)
+    return (
+      <UserContext.Provider
+        value={{
+          user: this.state.user,
+          setUser: this.setUser,
+          logout: this.logout,
+        }}
+      >
+        <Theme>{this.props.children}</Theme>
```

Comment:
Check out this `<Boot>` pattern for putting top-level stuff: https://github.com/artsy/palette/blob/master/packages/palette-docs/src/Boot.tsx

That way you can share boot across both server and client, while also having a nice place to store all of your context providers and other globals.

Reply:
And then you can point your SSR / Client config right at it:

- https://github.com/artsy/palette/blob/master/packages/palette-docs/gatsby-ssr.js#L10
- https://github.com/artsy/palette/blob/master/packages/palette-docs/gatsby-browser.js#L11

Reply:
Yeah i've called it Boot in the past but wasn't aware it was a pattern- `App` felt more semantically react but maybe `Boot` is more standard.

Reply:
"Boot" -> perform x operations in order to initialize the app, such as setting up providers and global state and theme and so on. This is where the application is booted. 

"App" -> "Boot" -> "Routes" -> "Route" -> "Page" -> "Component"